src/parser/mod.rs

```diff
@@ -4139,6 +4160,199 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse statements of the DropTrigger type such as:
+    ///
+    /// ```sql
+    /// DROP TRIGGER [ IF EXISTS ] name ON table_name [ CASCADE | RESTRICT ]
+    /// ```
+    pub fn parse_drop_trigger(&mut self) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
```

Comment:
```suggestion
        if !dialect_of!(self is PostgreSqlDialect | GenericDialect) {
              return self.expected(...)
        }
        ...
```
We can return early instead, so that the body of the function can be unnested, also we can include [genericdialect](https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#new-syntax) support as well

(same for create trigger)

Reply:
Applied the proposed change to both `create_trigger` and `drop_trigger`. Do let me know whether the change in commit ba5d47399067b6e4513befc661b48080844c9165 matches your expectations, so we may mark this conversation as resolved.